### PR TITLE
basic source implementation documentation

### PR DESCRIPTION
**Pull request by jrhizor**
This PR adds documentation for Singer and non-Singer sources and adds links on docs.airbyte.io.

The main thing we're missing for documentation is a https://github.com/singer-io/getting-started/blob/master/docs/SPEC.md equivalent for the Airbyte Protocol, which I'll be working on in a separate PR (and linking in these guides).

Once this process solidifies (easier testing, python-based testing, etc), it probably makes sense to do a video recording of the full end to end development process for a source.

I am also working on adding type hints and doc strings, but those changes are partially blocked by open restructuring PRs.

### CODE REVIEW

File: airbyte-integrations/template/singer-source/README.md [hunk 1]
```diff
@@ -1,4 +1,35 @@
...
+    1. Ultimately, configurations that match this schema will be generated by the Airbyte UI and passed to your integration as input. You can also manually generate config files and pass them to the integration CLI.
+1. `SOURCENAME_singer_source/source.py` is where you actually implement your integration.
+    1. `check` should check if a connection can be established to a source given a config that matches your specification.
+    1. `discover_cmd` returns a command string we will execute to retrieve a Singer catalog.
+        1. Behind the scenes, we convert this Singer catalog to an `AirbyteCatalog`. 
+        1. In some cases, your Singer Tap might not implement discover. In this case, you will have to add functionality to create your own catalog.
+    1. `read_cmd` will write messages to stdout. 
+        1. The command to specify is typically the same as the sync from a Singer Tap.
+        1. Usually this command needs to be computed from the config file and optionally the state.
+1. Update the `sample_files` directory with an example config and catalog (discover output).
```
Comment (cgardens) [inline airbyte-integrations/template/singer-source/README.md:24]: why is this necessary?

File: airbyte-integrations/template/python-source/README.md [hunk 2]
```diff
@@ -1,3 +1,32 @@
+### Create a Python Airbyte Source
+1. If you are trying to create an Airbyte Source from a Singer Tap, you should follow [this guide](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/template/singer-source/README.md) instead.
+1. Copy the template into a directory with your integration name. `cp -r airbyte-integrations/template/python-source airbyte-integrations/source-SOURCENAME`.
+1. ```
+   mv airbyte-integrations/source-SOURCENAME/template_python_source airbyte-integrations/source-SOURCENAME/SOURCENAME_source
+   ``` 
+1. Update the name, description, and dependencies in `setup.py`
+    1. Note that release dependencies should go in `setup.py`, NOT `requirements.txt`. The requirements file is only used for local development.
```
Comment (cgardens) [inline airbyte-integrations/template/python-source/README.md:8]: dumb question: if i have a 3rd party dep i need, do i need to put it in setup.py AND requirements.txt or is just putting it setup.py enough?

**Review by cgardens — APPROVED** [review]
i dig it.

File: airbyte-integrations/template/singer-source/README.md (hunk 1 above)
Comment (jrhizor) [reply airbyte-integrations/template/singer-source/README.md:24]: I'm adding this under the `README.md` request as
```
    1. We recommend including sample configs and catalogs in the `sample_files` directory or within your `README.md`.
```

It's not necessary, just recommended so it's easy to work with.

File: airbyte-integrations/template/python-source/README.md (hunk 2 above)
Comment (jrhizor) [reply airbyte-integrations/template/python-source/README.md]: Updated to make it clear that all of your dependencies should go in setup.py and that requirements.txt is reserved for internal Airbyte dependencies.

File: airbyte-integrations/template/python-source/README.md [hunk 3]
```diff
@@ -1,3 +1,32 @@
+### Create a Python Airbyte Source
+1. If you are trying to create an Airbyte Source from a Singer Tap, you should follow [this guide](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/template/singer-source/README.md) instead.
+1. Copy the template into a directory with your integration name. `cp -r airbyte-integrations/template/python-source airbyte-integrations/source-SOURCENAME`.
+1. ```
+   mv airbyte-integrations/source-SOURCENAME/template_python_source airbyte-integrations/source-SOURCENAME/SOURCENAME_source
```
Comment (michel-tricot) [inline airbyte-integrations/template/python-source/README.md:5]: Can you update to follow the new directory structure?

File: airbyte-integrations/template/python-source/README.md [hunk 4]
```diff
@@ -1,3 +1,32 @@
...
+   ``` 
+1. Update the name, description, and dependencies in `setup.py`
+    1. Note that all of your dependencies should go in `setup.py`, NOT `requirements.txt`. The requirements file is only used to connect internal Airbyte dependencies for local development. 
+1. Update the `Dockerfile` to point `CODE_PATH` and `AIRBYTE_IMPL_MODULE` at `SOURCENAME_source`
+1. Edit the classname in `SOURCENAME_singer_source/source.py` and update the `Dockerfile` to point `AIRBYTE_IMPL_PATH` to that classname.
```
Comment (michel-tricot) [inline airbyte-integrations/template/python-source/README.md:10]: Remove singer

File: airbyte-integrations/template/python-source/README.md (hunk 3 above)
Comment (jrhizor) [reply airbyte-integrations/template/python-source/README.md:5]: discussed offline, will need to happen as part of https://github.com/airbytehq/airbyte/pull/607 when it's merged to master.
